### PR TITLE
Introduced NetworkDataProvider and added tests for NetworkOperator

### PR DESCRIPTION
**Pull request by lennet**
Initially the NetworkOperator was just an abstraction for URLSessions but it gained some logic over the time so I introduced another abstraction layer to be able to mock network request without replacing a lot of core logic.

This allows us to write better test and read data from files for debugging purposes instead from a network connection (related to #141 ).

Test coverage for UnitTests increased from 26% to 30%

We could think about removing the NetworkLayer protocol as it isn't needed anymore because  test mocking is also possible with injecting a custom NetworkDataProvider.

### CODE REVIEW

File: CriticalMass/Constants.swift [hunk 1]
```diff
@@ -8,7 +8,6 @@
 import Foundation
 
 enum Constants {
-    static let twitterEndpoint = URL(string: "https://api.criticalmaps.net/twitter")!
```
Comment (lennet) [inline CriticalMass/Constants.swift]: This is not really related to the PR. Just some small cleanup of clean code

File: CriticalMass/NetworkOperator.swift [hunk 2]
```diff
@@ -57,7 +46,7 @@ struct NetworkOperator: NetworkLayer {
     private func dataTask(with request: URLRequest,
                           completion: @escaping ResultCallback<Data>) {
         networkIndicatorHelper.didStartRequest()
```
Comment (lennet) [inline CriticalMass/NetworkOperator.swift]: The new tests aren't covering networkIndicator activation. Not sure if tests for that are needed though

File: CriticalMass/AppController.swift [hunk 3]
```diff
@@ -26,11 +26,15 @@ class AppController {
...
 
     private lazy var requestManager: RequestManager = {
-        RequestManager(dataStore: dataStore, locationProvider: LocationManager(), networkLayer: networkOperator, idProvider: idProvider, url: Constants.apiEndpoint)
+        RequestManager(dataStore: dataStore, locationProvider: LocationManager(), networkLayer: networkOperator, idProvider: idProvider)
     }()
 
     private let networkOperator: NetworkOperator = {
-        NetworkOperator(networkIndicatorHelper: NetworkActivityIndicatorHelper())
+        let configuration = URLSessionConfiguration.default
+        configuration.requestCachePolicy = .reloadIgnoringLocalCacheData
```
Comment (mltbnz) [inline CriticalMass/AppController.swift]: While adding an error view I thought we might need to set a timeout timeinterval as well. But since it is not related to this PR not requested here. Just thought we could discuss here

File: CriticalMass/AppController.swift [hunk 4]
```diff
@@ -44,7 +48,7 @@ class AppController {
     }()
 
     private lazy var twitterManager: TwitterManager = {
-        TwitterManager(networkLayer: networkOperator, url: Constants.twitterEndpoint)
+        TwitterManager(networkLayer: networkOperator, request: TwitterRequest())
```
Comment (mltbnz) [inline CriticalMass/AppController.swift]: good idea

File: CriticalMass/AppController.swift (hunk 3 above)
Comment (lennet) [reply CriticalMass/AppController.swift]: Doesn’t the default configuration come with a timeout?

**Comment by lennet** [comment]
> Looks good. Might cause some trouble in my ErrorState PR but selber schuld :D

I’m okay with waiting with this PR until the ErrorState one got merged

File: CriticalMass/AppController.swift (hunk 3 above)
Comment (mltbnz) [reply CriticalMass/AppController.swift]: No. E.g the Twitter Feed was loading all the time when the BE was not responding
